tubular/dates.py

```diff
@@ -663,3 +663,344 @@ def transform(self, X):
+
+        if "timeofday" in include:
+            self.timeofday_mapping = {
+                vi: k for k, v in timeofday_mapping.items() for vi in v
+            }
+            if set(self.timeofday_mapping.keys()) != set(range(24)):
+                raise ValueError(
+                    "{}: timeofday mapping dictionary should contain mapping for all hours between 0-23. {} are missing".format(
+                        self.classname(),
+                        set(range(24)) - set(self.timeofday_mapping.keys()),
+                    )
+                )
+            # Check if all hours in dictionary
+
+        if "timeofmonth" in include:
+            self.timeofmonth_mapping = {
+                vi: k for k, v in timeofmonth_mapping.items() for vi in v
+            }
+            if set(self.timeofmonth_mapping.keys()) != set(range(32)):
+                raise ValueError(
+                    "{}: timeofmonth mapping dictionary should contain mapping for all days between 1-31. {} are missing".format(
+                        self.classname(),
+                        set(range(1, 32)) - set(self.timeofmonth_mapping.keys()),
+                    )
+                )
+
+        if "timeofyear" in include:
+            self.timeofyear_mapping = {
+                vi: k for k, v in timeofyear_mapping.items() for vi in v
+            }
+            if set(self.timeofyear_mapping.keys()) != set(range(1, 13)):
+                raise ValueError(
+                    "{}: timeofyear mapping dictionary should contain mapping for all months between 1-12. {} are missing".format(
+                        self.classname(),
+                        set(range(1, 13)) - set(self.timeofyear_mapping.keys()),
+                    )
+                )
+
+        if "dayofweek" in include:
+            self.dayofweek_mapping = {
+                vi: k for k, v in dayofweek_mapping.items() for vi in v
+            }
+            if set(self.dayofweek_mapping.keys()) != set(range(7)):
+                raise ValueError(
+                    "{}: dayofweek mapping dictionary should contain mapping for all days between 0-6. {} are missing".format(
+                        self.classname(),
+                        set(range(7)) - set(self.dayofweek_mapping.keys()),
+                    )
+                )
+
+    def identify_timeofday(self, hour):
+        """Map timeofday from hour
+
+        Parameters
+        ----------
+        hour : numeric
+            Hour to map
+
+        Returns
+        -------
+        str : str
+            Mapped hour
+        """
+
+        if not type(hour) is float:
+            if not type(hour) is int:
+                raise TypeError(f"{self.classname()}: hour should be float or int")
+
+        if not np.isnan(hour):
+            if hour not in np.arange(0, 24, 1):
+                raise ValueError(
+                    f"{self.classname()}: hour should be a whole value in 0-23"
```

Comment:
This error isn't particularly descriptive given that it occurs a couple of levels deep in the transformers operation

This also applies to all of the identify methods

Reply:
Also I think "..an integer value (or float representation thereof) in the range 0-23" would read nicer than whole value -- (also not quite sure that this the appropriate use of thereof...)

Reply:
addressed in https://github.com/lvgig/tubular/pull/49/commits/c98c4ecc40a1ac706145f4aeb5f58970d91a7a60